packages/core/test/Form.test.jsx

```diff
@@ -4013,6 +4013,41 @@ describe('Form omitExtraData and liveOmit', () => {
       // We use setTimeout with a delay of 0ms to allow all asynchronous operations to complete in the React component.
       // Despite this being a workaround, it turned out to be the only effective method to handle this test case.
     });
+
+    it('should reset when schema changes', () => {
+      const schema = {
+        type: 'object',
+        properties: {
+          foo: { type: 'string' },
+        },
+      };
+
+      const extraErrors = {
+        foo: {
+          __errors: ['foo'],
+        },
+      };
```

Comment:
Rather than setting extra errors, might I suggest making foo required, not providing a form value, clicking submit, seeing that there is an error, and changing it to be optional and then verifying the error was cleared?